### DISCUSSION TITLE
Group resizing logic

### DISCUSSION THREAD

**Discussion by ShaMan123** [fabricjs/fabric.js#7136] [Ideas]
After using `Group` a lot the need to control it's size has become the main thing I have against it.
Setting `width` | `height` is irrelevant after adding/removing an object because it recalculates it's dimensions with `_calcBounds`.
I find myself extending group all the time because of this.
This is bad and is inconsistent with the rest of the shapes.
It is tough to handle for someone starting to use fabric and for others as well. It is very internal on the one hand and very unintuitive on the other. I lost a good amount of hours understanding this.

Ideas:
1. Adding a prop `resizeOnAddRemove` similar to `renderOnAddRemove`. It can default to true so it won't be breaking. The problem is we need a prop for positioning as well I think because sometimes group does that to.
2. Exposing some method on group similar to `onSelect` to determine if and how it should resize:
```ts
onBoundsChange(left, top, width, height) {
// return the values you want the group to set, true to resize, false not to??
}
```
3. Adding an options param to `addWithUpdate`, `removeWithUpdate`: `addWithUpdate(object, options: {resize:boolean, reposition:boolean})`

@asturur @melchiar

**Comment by asturur** [comment]
So my first question is, why do you need to change the group size? what is the need behind it?

**Reply by ShaMan123** [reply]
Say I have a group with some objects. Now I want to add another object to it but I don't want it to resize if the object is bigger than the group. For example I draw with PencilBrush over that group and I want the created path to be clipped to the group. Controlling resizing allows me to do that without handling complex logic.

**Reply by ShaMan123** [reply]
Another use case is a drop zone group for objects. Once you drop them there they are added to the drop zone group but you want the drop zone to remain the same size.

**Reply by asturur** [reply]
I know this is 2 weeks old but there are so many things going on here on the repo those days.
So why group.add and group.addWithUpdate do not already cover your usecases?
one add without resizing, the other does.

**Reply by ShaMan123** [reply]
I guess you are right. 
Thanks

**Comment by ShaMan123** [comment]
#7449
